cwms-data-api/src/main/java/cwms/cda/api/BlobController.java

```diff
@@ -130,21 +130,33 @@ public void getAll(@NotNull Context ctx) {
             description = "Returns the binary value of the requested blob as a seekable stream with the "
                     + "appropriate media type.",
             queryParams = {
-                    @OpenApiParam(name = OFFICE, description = "Specifies the owning office."),
+                @OpenApiParam(name = OFFICE, description = "Specifies the owning office."),
+                @OpenApiParam(name = BLOB_ID, description = "If this _query_ parameter is provided the id _path_ parameter "
+                    + "is ignored and the value of the query parameter is used.   "
+                    + "Note: this query parameter is necessary for id's that contain '/' or other special "
+                    + "characters.  Because of abuse even properly escaped '/' in url paths are blocked.  "
```

Comment:
It's not because of abuse. It's because of how Javalin handles the path parameter. In theory it's *supposed* to handle it, but it doesn't so do correctly.

Another approach I was thinking of taking (which I think would be complimentary to this one) was to base64 encode the id for the path parameter. Not easy for the user to do, but not difficult for cwms-python/cwms-cli.